blog/integration-tests-with-ranger.md

```diff
@@ -0,0 +1,334 @@
+generate 50 objects of Measurement class such that `owner` is someone from this set ("alex", "bob", "charlie"), and `created` is a timestamp between [1496300000, 196400000), and `sensor` can be something from this set ("heart-rate-monitor", "accelerometer", "hygrometer", "thermometer") and so on. The recipe looks like this:
+
+```java
+ObjectGenerator<Measurement> measurementGenerator = 
+  new ObjectGenerator.Builder<Measurement>(Measurement.class)
+    .withValues("owner", "alex", "bob", "charlie")
+    .withValues("created", 1496300000L, 196400000L)
+    .withValues("sensor",  "heart-rate-monitor", "accelerometer", "hygrometer", "thermometer")
+    .toBeGenerated(50)
+    .build();
+List<Measurement> measurements = measurementGenerator.generate();
+```
+
+Simple, isn't it? We can also combine the recipes so we can make data abide certain rules like: create a thousand Measurement objects for a bunch of users and with bunch of sensors, but only 100 of them will be from user "charlie". In other words, we want 900 measurement object for non-charlie users and 100 for user "charlie":
+
+```java
+ObjectGenerator<Measurement> nonCharlieMeasurementGenerator = 
+    new ObjectGenerator.Builder<Measurement>(Measurement.class)
+    .withValues("owner", "alex", "bob", "david", "emma")
+    .withValues("created", 1496300000L, 196400000L)
+    .withValues("sensor",  "heart-rate-monitor", "accelerometer", "hygrometer", "thermometer")
+    .toBeGenerated(900)
+    .build();
+
+ObjectGenerator<Measurement> charlieMeasurementGenerator = 
+    new ObjectGenerator.Builder<Measurement>(Measurement.class)
+    .withValues("owner", "charlie")
+    .withValues("created", 1496300000L, 196400000L)
+    .withValues("sensor",  "heart-rate-monitor", "accelerometer", "hygrometer", "thermometer")
+    .toBeGenerated(100)
+    .build();
+```
+
+And now we just pass `nonCharlieMeasurementGenerator` and `charlieMeasurementGenerator` to `AggregatedObjectGenerator<Measurement>` in order to generate objects from both generators:
+
+```java
+AggregatedObjectGenerator<Measurement> aggregatedObjectGenerator = 
+    new AggregatedObjectGenerator.Builder<Measurement>()
+    .withObjectGenerator(nonCharlieMeasurementGenerator)
+    .withObjectGenerator(charlieMeasurementGenerator)
+    .build();
+
+List<Measurement> measurements = aggregatedObjectGenerator.generateAll();
+```
+
+Voilà!
+
+## Ranger - real world example
+
+Now that we are familiar with the basic building blocks for Ranger library, let's try it out with a real world example.
+
+I worked on a IoT application that collects measurements from bunch of sensors and provides analytics and alerting based on the stored data. And there was a database query for fetching the newest `n` results for a particular user and sensor. MongoDB was used for storage. So I had the mongodb query that looks something like this:
```

Comment:
You could say "I used to work on a IoT..." to emphasize that you don't work on it anymore. "I worked on..." sounds like an answer to the question "Who worked on it?"